fix typos in chapter 5 (#38)
soupi/learn-haskell-blog-generator: src/05-glue/01-markup_to_html.md

If we would've imported `Html.Internal` qualified, we'd have to write
`Html.Internal.el` which is a bit long.

> We can also give a new name to the module with the `as` keyword:

Reviewer: ```suggestion
We can also give the module a new name with the `as` keyword:
```

```hs
import qualified Html.Internal as HI
```

And write `HI.el` instead.

I like using qualified imports because readers do not have to guess where a
name comes from. Some modules are even designed to be imported qualified.
For example, many container APIs such as maps, sets, and vectors have very similar

> API. If we need multiple containers in a single module we pretty much have

Reviewer: ```suggestion
API. If we want to use multiple containers in a single module we pretty much have
```

to use qualified imports so that when we write a function such as `singleton`,
which creates a container with a single value, GHC will know which `singleton`
function we are referring to.

Some people prefer to use import lists instead of qualified imports,
because qualified names can be a bit verbose and noisy. I usually prefer them.

> For more information about imports,

Reviewer: ```suggestion
Some people prefer to use import lists instead of qualified imports,
because qualified names can be a bit verbose and noisy.
I will often prefer qualified imports to import lists, but feel free to
try both solutions and see which fits you better.
For more information about imports,
```

headings that are not `h1`. There are a few ways to handle this:

- Ignore the warning - this will likely fail at runtime one day and the user will be sad
- Pattern match other cases and add a nice error with the `error` function - it has
  the same disadvantage above, but will also not notify any issue

>   here at compile time.

Reviewer: ```suggestion
  the same disadvantage above, but will also no longer notify of the unhandled
  cases at compile time.
```

```

> To concatenate all of the `Html.Structure`, we could write a recursive

Reviewer: ```suggestion
To concatenate all of the `Html.Structure`, we could try to write a recursive
```

---

Remember the `<>` function we implemented as an instance of the `Semigroup`

> type class? `Semigroup` is an **abstraction** for things

Reviewer: ```suggestion
type class? We mentioned that `Semigroup` is an **abstraction** for things
```

that implements `(<>) :: a -> a -> a`, where  `<>` is associative
(`a <> (b <> c) = (a <> b) <> c`).

It turns out to be a common pattern to have an instance of `Semigroup` with an "empty" value.
For example a string.

> And this is actually a well known **abstraction** called **monoid**.

Reviewer: ```suggestion
It turns out that having an instance of `Semigroup` and also having a value that represents
an "empty" value is a fairly common pattern. For example a string can be concatenated, 
and the empty string can serve as an "empty" value.
And this is actually a well known **abstraction** called **monoid**.
```

    x : xs -> x <> mconcat xs
```

Notice that because `Semigroup` is a *super class* of `Monoid`,
we can still use the `<>` function from the `Semigroup` class
without adding the `Semigroup a` constraint to the left side of `=>`.
By adding the `Monoid a` constraint we implicitly add a `Semigroup a`
constraint as well!

This `mconcat` function is very similar to the `concatStructure` function,
but this one works for any `Monoid`, including `Structure`!

> Abstractions help us **reuse** code!

Reviewer: ```suggestion
Abstractions help us identify common patterns and **reuse** code!
```
